### DISCUSSION TITLE
Can we build Android Extensions that modify the context menu?

### DISCUSSION THREAD

**Discussion by madCode** [mdn/mdn-community#473] [Platform] [answered]
I'm trying to build an extension for the Android version of Firefox that adds a custom option to the context menu that pops up when you select text. However, I'm having difficulty getting any button to show up. I tried installing [this example](https://github.com/mdn/webextensions-examples/tree/main/context-menu-copy-link-with-types) with no changes on my Firefox Night for Developers app. I see the extension in the app, but no "Copy link to clipboard" button appears in the context menu. Is this a known limitation of Firefox's Android app or am I missing something basic?

**Comment by Rumyra** [comment]
Mentioning @mdn/yari-content-mozilla-add-ons who may be able to provide insight here

**Comment by willdurand** [comment]
We do not support this API on Firefox for Android.

**Reply by BuddhaDiedLaughing** [reply]
AS A USER THAT IS ANGERING.  When will Mozilla stop pacing Firefox awesomeness to Chrome?  It's as if the mentality is that there's no need to advance that much further than Chromium browsers.  Same goes for the abysmal Firefox tabs UI,  native translation, autofill & the apparent depreciation of copy link text on mobile which is where it's most useful!

**Comment by Rob--W** [comment, marked as answer]
> We do not support this API on Firefox for Android.

If we ever decide to support it, then there would be activity at https://bugzilla.mozilla.org/show_bug.cgi?id=1595822

Your currently available option is to use add a browserAction button (or simply called action in MV3) to your extension and use scripting.executeScript + `window.getSelection()` to get the current text selection (assuming that it is in the top-level document), and then the navigator.clipboard or document.execCommand API to do the actual copy.

**Reply by Rumyra** [reply]
Thank you @Rob--W !
